homeassistant/components/scrape/sensor.py

```diff
@@ -97,45 +98,47 @@ async def async_setup_platform(
         rest = create_rest_data_from_config(hass, resource_config)
 
         coordinator = ScrapeCoordinator(hass, rest, SCAN_INTERVAL)
-        await coordinator.async_refresh()
-        if coordinator.data is None:
-            raise PlatformNotReady
 
-        sensor_config = config
-        template_config = vol.Schema(
-            TEMPLATE_SENSOR_BASE_SCHEMA.schema, extra=vol.REMOVE_EXTRA
-        )(sensor_config)
+        sensors_config: list[tuple[ConfigType, ConfigType]] = [
+            (
+                config,
+                vol.Schema(TEMPLATE_SENSOR_BASE_SCHEMA.schema, extra=vol.REMOVE_EXTRA)(
+                    config
+                ),
+            )
+        ]
 
     else:
         coordinator = discovery_info["coordinator"]
-        sensor_config = discovery_info["config"]
-        template_config = sensor_config
+        sensors_config = [
+            (sensor_config, sensor_config)
+            for sensor_config in discovery_info["configs"]
+        ]
 
-    name: str = template_config[CONF_NAME]
-    unique_id: str | None = template_config.get(CONF_UNIQUE_ID)
-    select: str | None = sensor_config.get(CONF_SELECT)
-    attr: str | None = sensor_config.get(CONF_ATTRIBUTE)
-    index: int = sensor_config[CONF_INDEX]
-    value_template: Template | None = sensor_config.get(CONF_VALUE_TEMPLATE)
+    await coordinator.async_refresh()
+    if coordinator.data is None:
+        raise PlatformNotReady
 
-    if value_template is not None:
-        value_template.hass = hass
+    for sensor_config, template_config in sensors_config:
```

Comment:
I'm not following the function of the tuple of configs. Why do we need both the raw and the validated config?

Reply:
It's because we have the "legacy" schema that we are not allowed to change according to ADR0007
`vol.Schema(TEMPLATE_SENSOR_BASE_SCHEMA.schema, extra=vol.REMOVE_EXTRA)`

Maybe if we change it to `vol.Schema(TEMPLATE_SENSOR_BASE_SCHEMA.schema, extra=vol.ALL_EXTRA)` we don't need to keep the tuple...

Reply:
I've tested it locally - it works with ALLOW_EXTRA

Reply:
```suggestion
    entities: list[ScrapeSensor] = []
    for sensor_config in sensors_config:
```

Reply:
Ok, I think I get it now.